etld/data/build.js

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env node
+
+/* Copyright (c) 2019 The Brave Software Team. Distributed under the MPL2
+ * license. This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This code parses the public suffix rule list into a header file of
+ * rules that gets build into the ad-block library.
+ */
+
+const fsLib = require("fs");
+const pathLib = require("path");
+
+const ruleTextPath = pathLib.join(__dirname, "public_suffix_list.dat");
+const ruleText = fsLib.readFileSync(ruleTextPath, "utf8");
+
+const templatePath = pathLib.join(__dirname, "public_suffix_list.h.template");
+const templateText = fsLib.readFileSync(templatePath, "utf8");
+const rules = [];
+
+const toPublicSuffixRuleSerializedString = (isWildcard, isException, labels) => {
+  const constructorArgs = [];
+  constructorArgs.push(isWildcard ? "true" : "false");
+  constructorArgs.push(isException ? "true" : "false");
+
+  const wrappedLabels = labels.map(JSON.stringify);
+  constructorArgs.push("{" + wrappedLabels.join(", ") + "}");
```

Comment:
Should the labels be surrounded by `"` in case they're not all bare words? Also, is it possible they include `"` characters that should be escaped?